src/Translation/Manager.php

```diff
@@ -243,6 +264,10 @@ public function translate($key, $lang = null)
                 $this->addNotFound($key, $lang);
             }
 
+            if($key === $translated) {
+                $this->logger->warning('Translation not found', array('key' => $key, 'lang' => $lang));
```

Comment:
On my side, a translation can not exist in specific language and let fallback on default, so a missing translation key can be the right behavior. So if you don't support that, yeah it have sense to log as warning.